perses/dispersed/feptasks.py

```diff
@@ -31,41 +32,67 @@
 EquilibriumResult = NamedTuple('EquilibriumResult', [('sampler_state', states.SamplerState), ('reduced_potential', float)])
 NonequilibriumResult = NamedTuple('NonequilibriumResult', [('cumulative_work', np.array)])
 
+class NoTrajectoryException(Exception):
+    pass
+
 class NonequilibriumSwitchingMove(mcmc.BaseIntegratorMove):
     """
     This class represents an MCMove that runs a nonequilibrium switching protocol using the AlchemicalNonequilibriumLangevinIntegrator.
     It is simply a wrapper around the aforementioned integrator, which must be provided in the constructor.
 
     Parameters
     ----------
-    n_steps : int
-        The number of integration steps to take each time the move is applied.
     integrator : openmmtools.integrators.AlchemicalNonequilibriumLangevinIntegrator
         The integrator that will be used for the nonequilibrium switching.
+    work_save_interval : int, default None
+        The frequency with which to record the cumulative total work. If None, only save the total work at the end
+    top: md.Topology, default None
+        The topology to use to write the positions along the protocol. If None, don't write anything.
+    subset_atoms : np.array, default None
+        The indices of the subset of atoms to write. If None, write all atoms (if writing is enabled)
     context_cache : openmmtools.cache.ContextCache, optional
         The ContextCache to use for Context creation. If None, the global cache
         openmmtools.cache.global_context_cache is used (default is None).
-    reassign_velocities : bool, optional
-        If True, the velocities will be reassigned from the Maxwell-Boltzmann
-        distribution at the beginning of the move (default is False).
-    restart_attempts : int, optional
-        When greater than 0, if after the integration there are NaNs in energies,
-        the move will restart. When the integrator has a random component, this
-        may help recovering. An IntegratorMoveError is raised after the given
-        number of attempts if there are still NaNs.
-
     Attributes
     ----------
-    n_steps : int
-    context_cache : openmmtools.cache.ContextCache
-    reassign_velocities : bool
-    restart_attempts : int or None
     current_total_work : float
+        The total work in kT.
     """
 
-    def __init__(self, integrator: integrators.AlchemicalNonequilibriumLangevinIntegrator, n_steps: int, **kwargs):
-        super(NonequilibriumSwitchingMove, self).__init__(n_steps, **kwargs)
-        self._integrator = integrator
+    def __init__(self, integrator: integrators.AlchemicalNonequilibriumLangevinIntegrator,
+        work_save_interval: int=None, top: md.Topology=None, subset_atoms: np.array=None, **kwargs):
+
+        super(NonequilibriumSwitchingMove, self).__init__(**kwargs)
+        self._integrator = copy.deepcopy(integrator)
```

Comment:
With this `copy`, I think you'll lose the integrator Python layer. You'll have to call `RestorableIntegrator.restore_interface(integrator)` to recover the Python methods, but you'll still lose the Python attributes.

I'm fixing this right now as part of the next feature I'm implementing. It should be ready by the end of the day, and I'd expect the new `openmmtools` release to come out by the end of the week/beginning of next week so, if you can wait a little longer, this won't need changes.

Reply:
Ok, thanks for the heads up.